Spell check error on multiple times error

<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this, make sure that the issue doesn't already exist
  or if it is not closed.

  Is your issue fixed on the preview release?: pip install pylint astroid --pre -U

-->

### Steps to reproduce
1. Write a docstring with some bad word two or more times in the same line
2. Run spell check `pylint --disable all --enable spelling --spelling-dict en_US test_spell.py`
3. See how the word is erroneously pointed 

### Current behavior
Bad pointed word after the first

### Expected behavior
Every word have to by pointer in thier respective position

### pylint --version output
pylint --version                                                            
pylint 2.3.1
astroid 2.2.5
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]

```
❯ cat test_spell.py 
def test():
    """
    teh teh
    hotsname hotsname
    teh import thing to teh pylint
    """
    pass

❯ pylint --disable all --enable spelling --spelling-dict en_US test_spell.py
************* Module test_spell
test_spell.py:3:0: C0402: Wrong spelling of a word 'teh' in a docstring:
    teh teh
    ^^^
Did you mean: ''the' or 'eh' or 'tech' or 'tee''? (wrong-spelling-in-docstring)
test_spell.py:3:0: C0402: Wrong spelling of a word 'teh' in a docstring:
    teh teh
    ^^^
Did you mean: ''the' or 'eh' or 'tech' or 'tee''? (wrong-spelling-in-docstring)
test_spell.py:4:0: C0402: Wrong spelling of a word 'hotsname' in a docstring:
    hotsname hotsname
    ^^^^^^^^
Did you mean: ''hots name' or 'hots-name' or 'whatshisname''? (wrong-spelling-in-docstring)
test_spell.py:4:0: C0402: Wrong spelling of a word 'hotsname' in a docstring:
    hotsname hotsname
    ^^^^^^^^
Did you mean: ''hots name' or 'hots-name' or 'whatshisname''? (wrong-spelling-in-docstring)
test_spell.py:5:0: C0402: Wrong spelling of a word 'teh' in a docstring:
    teh import thing to teh pylint
    ^^^
Did you mean: ''the' or 'eh' or 'tech' or 'tee''? (wrong-spelling-in-docstring)
test_spell.py:5:0: C0402: Wrong spelling of a word 'teh' in a docstring:
    teh import thing to teh pylint
    ^^^
Did you mean: ''the' or 'eh' or 'tech' or 'tee''? (wrong-spelling-in-docstring)

----------------------------------------------------------------------
Your code has been rated at -20.00/10 (previous run: -20.00/10, +0.00)

See how when the second word is checked, the first is pointed again.
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -299,3 +299,5 @@ contributors:
 
 * Zeb Nicholls: contributor
     - Made W9011 compatible with 'of' syntax in return types
+
+* Agustin Toledo: contributor

--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,10 @@ What's New in Pylint 2.4.0?
 
 Release date: TBA
 
+* Fix pointer on spelling check when the error are more than one time in the same line. 
+
+  Close #2895
+
 * Allow of in `GoogleDocstring.re_multiple_type`
 
 * Added `subprocess-run-check` to handle subrocess.run without explicitly set `check` keyword.

--- a/pylint/checkers/spelling.py
+++ b/pylint/checkers/spelling.py
@@ -285,7 +285,7 @@ class SpellingChecker(BaseTokenChecker):
     def _check_spelling(self, msgid, line, line_num):
         original_line = line
         try:
-            initial_space = re.search(r"[^\S]\s*", line).regs[0][1]
+            initial_space = re.search(r"^[^\S]\s*", line).regs[0][1]
         except (IndexError, AttributeError):
             initial_space = 0
         if line.strip().startswith("#"):

--- a/pylint/checkers/spelling.py
+++ b/pylint/checkers/spelling.py
@@ -284,12 +284,17 @@ class SpellingChecker(BaseTokenChecker):
 
     def _check_spelling(self, msgid, line, line_num):
         original_line = line
+        try:
+            initial_space = re.search(r"[^\S]\s*", line).regs[0][1]
+        except (IndexError, AttributeError):
+            initial_space = 0
         if line.strip().startswith("#"):
             line = line.strip()[1:]
             starts_with_comment = True
         else:
             starts_with_comment = False
-        for word, _ in self.tokenizer(line.strip()):
+        for word, word_start_at in self.tokenizer(line.strip()):
+            word_start_at += initial_space
             lower_cased_word = word.casefold()
 
             # Skip words from ignore list.
@@ -328,12 +333,15 @@ class SpellingChecker(BaseTokenChecker):
                 suggestions = self.spelling_dict.suggest(word)
                 del suggestions[self.config.max_spelling_suggestions :]
 
-                m = re.search(r"(\W|^)(%s)(\W|$)" % word, line)
+                line_segment = line[word_start_at:]
+                m = re.search(r"(\W|^)(%s)(\W|$)" % word, line_segment)
                 if m:
                     # Start position of second group in regex.
                     col = m.regs[2][0]
                 else:
-                    col = line.index(word)
+                    col = line_segment.index(word)
+
+                col += word_start_at
 
                 if starts_with_comment:
                     col += 1

--- a/pylint/test/unittest_checker_spelling.py
+++ b/pylint/test/unittest_checker_spelling.py
@@ -303,3 +303,61 @@ class TestSpellingChecker(CheckerTestCase):
             )
         ):
             self.checker.visit_classdef(stmt)
+
+    @skip_on_missing_package_or_dict
+    @set_config(spelling_dict=spell_dict)
+    def test_more_than_one_error_in_same_line_for_same_word_on_docstring(self):
+        stmt = astroid.extract_node(
+            'class ComentAbc(object):\n   """Check teh dummy comment teh"""\n   pass'
+        )
+        with self.assertAddsMessages(
+            Message(
+                "wrong-spelling-in-docstring",
+                line=2,
+                args=(
+                    "teh",
+                    "Check teh dummy comment teh",
+                    "      ^^^",
+                    self._get_msg_suggestions("teh"),
+                ),
+            ),
+            Message(
+                "wrong-spelling-in-docstring",
+                line=2,
+                args=(
+                    "teh",
+                    "Check teh dummy comment teh",
+                    "                        ^^^",
+                    self._get_msg_suggestions("teh"),
+                ),
+            )
+        ):
+            self.checker.visit_classdef(stmt)
+
+    @skip_on_missing_package_or_dict
+    @set_config(spelling_dict=spell_dict)
+    def test_more_than_one_error_in_same_line_for_same_word_on_comment(self):
+        with self.assertAddsMessages(
+            Message(
+                "wrong-spelling-in-comment",
+                line=1,
+                args=(
+                    "coment",
+                    "# bad coment coment",
+                    "      ^^^^^^",
+                    self._get_msg_suggestions("coment"),
+                ),
+            ),
+            Message(
+                "wrong-spelling-in-comment",
+                line=1,
+                args=(
+                    "coment",
+                    "# bad coment coment",
+                    "             ^^^^^^",
+                    self._get_msg_suggestions("coment"),
+                ),
+            )
+        ):
+            self.checker.process_tokens(_tokenize_str("# bad coment coment"))
+

--- a/pylint/test/unittest_checker_spelling.py
+++ b/pylint/test/unittest_checker_spelling.py
@@ -330,7 +330,7 @@ class TestSpellingChecker(CheckerTestCase):
                     "                        ^^^",
                     self._get_msg_suggestions("teh"),
                 ),
-            )
+            ),
         ):
             self.checker.visit_classdef(stmt)
 
@@ -357,7 +357,6 @@ class TestSpellingChecker(CheckerTestCase):
                     "             ^^^^^^",
                     self._get_msg_suggestions("coment"),
                 ),
-            )
+            ),
         ):
             self.checker.process_tokens(_tokenize_str("# bad coment coment"))
-

